### PR TITLE
Escape implicit backslashes

### PR DESCRIPTION
**Pull request by MathiasReker**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Escape implicit backslashes
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | /
| How to test?  | /

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11665)

### CODE REVIEW

**Comment by mickaelandrieu** [comment]
Sadly, I'm not really good at string/regex stuff :lollipop:,

but we have @PierreRambaud who is really good at :+1:

File: classes/Tools.php [hunk 1]
```diff
@@ -2542,7 +2542,7 @@ public static function generateHtaccess($path = null, $rewrite_settings = null,
         fwrite($write_fd, "AddType application/font-woff .woff\n");
         fwrite($write_fd, "AddType font/woff2 .woff2\n");
         fwrite($write_fd, "<IfModule mod_headers.c>
-	<FilesMatch \"\.(ttf|ttc|otf|eot|woff|woff2|svg)$\">
+	<FilesMatch \"\\.(ttf|ttc|otf|eot|woff|woff2|svg)$\">
```
Comment (PierreRambaud) [inline classes/Tools.php:2545]: Nope, otherwise a file named `thisiseot` will match. You need to revert this.

File: classes/Translate.php [hunk 2]
```diff
@@ -35,7 +35,7 @@ public static function getFrontTranslation($string, $class, $addslashes = false,
     {
         global $_LANG;
 
-        $string = preg_replace("/\\\*'/", "\'", $string);
+        $string = preg_replace("/\\\\*'/", "\\'", $string);
```
Comment (PierreRambaud) [inline classes/Translate.php]: Backslashes were already escape

File: classes/Translate.php [hunk 3]
```diff
@@ -99,7 +99,7 @@ public static function getAdminTranslation($string, $class = 'AdminTab', $addsla
             }
         }
 
-        $string = preg_replace("/\\\*'/", "\'", $string);
+        $string = preg_replace("/\\\\*'/", "\\'", $string);
```
Comment (PierreRambaud) [inline classes/Translate.php]: Backslashes were already escape

File: admin-dev/filemanager/include/utils.php [hunk 4]
```diff
@@ -176,7 +176,7 @@ function fix_filename($str, $transliteration)
             $str = iconv('UTF-8', 'ASCII//TRANSLIT//IGNORE', $str);
         }
 
-        $str = preg_replace("/[^a-zA-Z0-9\.\[\]_| -]/", '', $str);
+        $str = preg_replace("/[^a-zA-Z0-9\\.\\[\\]_| -]/", '', $str);
```
Comment (PierreRambaud) [inline admin-dev/filemanager/include/utils.php:179]: Can you provide example of data?
You break the `[`, `.`, `]` and the `\` escapes.

File: classes/Translate.php (hunk 2 above)
Comment (PierreRambaud) [inline classes/Translate.php:38]: Same here can you provide example?

File: classes/db/DbPDO.php [hunk 5]
```diff
@@ -292,7 +292,7 @@ public function getVersion()
     public function _escape($str)
     {
         $search = array('\\', "\0", "\n", "\r", "\x1a", "'", '"');
-        $replace = array('\\\\', '\\0', '\\n', '\\r', "\Z", "\'", '\"');
+        $replace = array('\\\\', '\\0', '\\n', '\\r', '\\Z', "\\'", '\"');
```
Comment (PierreRambaud) [inline classes/db/DbPDO.php:295]: `\'`was already ok. Same for Z :)

File: install-dev/upgrade/php/update_customer_default_group.php [hunk 6]
```diff
@@ -31,7 +31,7 @@ function update_customer_default_group()
     copy($filename, $filename_old);
     chmod($filename_old, 0664);
     $content = file_get_contents($filename);
-    $pattern = "/define\('_PS_DEFAULT_CUSTOMER_GROUP_', (\d)\);/";
+    $pattern = "/define\\('_PS_DEFAULT_CUSTOMER_GROUP_', (\\d)\\);/";
```
Comment (PierreRambaud) [inline install-dev/upgrade/php/update_customer_default_group.php:34]: Nope you break the regex

File: tools/build/Library/ReleaseCreator.php [hunk 7]
```diff
@@ -709,8 +709,8 @@ protected function createZipArchive()
         $argTempProjectPath = escapeshellarg($this->tempProjectPath);
         $argInstallerZipFilename = escapeshellarg($installerZipFilename);
         $argProjectPath = escapeshellarg($this->projectPath);
-        $cmd = "cd {$argTempProjectPath} \
-            && zip -rq {$argInstallerZipFilename} . \
+        $cmd = "cd {$argTempProjectPath} \\
```
Comment (PierreRambaud) [inline tools/build/Library/ReleaseCreator.php:712]: You break the cli command.

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
We can approve since you maybe add a script to replace single `\`  by `\\` :/

**Comment by MathiasReker** [comment]
Sorry, @PierreRambaud. My script is apparently bugged.

**Comment by MathiasReker** [comment]
If some changed should stay, please let me know.

**Comment by PierreRambaud** [comment]
No problem @MathiasReker We really appreciate what you are doing for this project :smile: 
And hope you find the time to sleep :wink:

**Pull request closed without merging**